### ISSUE TITLE
Running on AWS Linux inside a private subnet in a VPC

### ISSUE THREAD

**Issue by oli-logicnow** [aweber/rabbitmq-autocluster#102]
Hi,
I'm struggling to get a cluster to form using the aws backend and making use of the autoscaling group with 3 instances.  I've battled with this for a few days and can get a cluster to form manually just fine on the same instances, so I'm pretty confident I've got ports/hostnames sorted, however I'm yet to have a cluster form with autocluster.  

I'll happily Paypal the cash for a beverage of your choice if you can point me in the right direction on this one.

**The setup**
Apologies for the info bomb here, just trying to cover all bases.

3 Instances, all launching from an AMI which is based on Amazon Linux AMI 2016.09.0 (HVM), into a private subnet inside our VPC, with both DNS hostname and DNS resolution set to yes.

UserData pushes in the following to set the hostname to to long form (ip-10-0-1-199.eu-west-1.compute.internal for example) and then start rabbit

```
#!/bin/bash
hostname $(hostname -f)
service rabbitmq-server start
```

All instances are the same, running RabbitMQ 3.6.2 on Erlang 19.1.3

/etc/rabbitmq/rabbitmq.config
```
[
  {rabbit, [
    {log_levels, [{autocluster, info}, {connection, info}]}
  ]},
  {autocluster, [
    {backend, aws},
    {aws_autoscaling, true}
  ]}
].
```

/etc/rabbitmq/rabbitmq-env.conf
```
RABBITMQ_USE_LONGNAME=true
RABBITMQ_DIST_PORT=25672
RABBITMQ_SERVER_ERL_ARGS="+K true +A128 +P 1048576 -kernel inet_default_connect_options [{nodelay,true}]
```

**The results**

The first instance to boot has the following 2 logs

rabbit@ip-10-0-1-183.eu-west-1.compute.internal.log
```
=INFO REPORT==== 21-Oct-2016::13:14:50 ===
Starting RabbitMQ 3.6.2 on Erlang 19.1.3
Copyright (C) 2007-2016 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 21-Oct-2016::13:14:50 ===
node           : rabbit@ip-10-0-1-183.eu-west-1.compute.internal
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
cookie hash    : jG/hKlMkavn9eqi0wsTy/A==
log            : /var/log/rabbitmq/rabbit@ip-10-0-1-183.eu-west-1.compute.internal.log
sasl log       : /var/log/rabbitmq/rabbit@ip-10-0-1-183.eu-west-1.compute.internal-sasl.log
database dir   : /var/lib/rabbitmq/mnesia/rabbit@ip-10-0-1-183.eu-west-1.compute.internal

=INFO REPORT==== 21-Oct-2016::13:14:53 ===
autocluster: Delaying startup for 2568ms.

=INFO REPORT==== 21-Oct-2016::13:14:55 ===
autocluster: Starting aws registration.

=INFO REPORT==== 21-Oct-2016::13:14:56 ===
Error description:
   {could_not_start,rabbit,
       {function_clause,
           [{autocluster,maybe_register,
                [error,aws,autocluster_aws],
                [{file,"src/autocluster.erl"},{line,111}]},
            {autocluster,init,0,[{file,"src/autocluster.erl"},{line,33}]},
            {rabbit_boot_steps,'-run_step/2-lc$^1/1-1-',1,
                [{file,"src/rabbit_boot_steps.erl"},{line,49}]},
            {rabbit_boot_steps,run_step,2,
                [{file,"src/rabbit_boot_steps.erl"},{line,49}]},
            {rabbit_boot_steps,'-run_boot_steps/1-lc$^0/1-0-',1,
                [{file,"src/rabbit_boot_steps.erl"},{line,26}]},
            {rabbit_boot_steps,run_boot_steps,1,
                [{file,"src/rabbit_boot_steps.erl"},{line,26}]},
            {rabbit,start,2,[{file,"src/rabbit.erl"},{line,585}]},
            {application_master,start_it_old,4,
                [{file,"application_master.erl"},{line,273}]}]}}

Log files (may contain more information):
   /var/log/rabbitmq/rabbit@ip-10-0-1-183.eu-west-1.compute.internal.log
   /var/log/rabbitmq/rabbit@ip-10-0-1-183.eu-west-1.compute.internal-sasl.log

```

rabbit@ip-10-0-1-183.eu-west-1.compute.internal-sasl.log
```
=CRASH REPORT==== 21-Oct-2016::13:14:56 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.151.0>
    registered_name: []
    exception exit: {bad_return,
                        {{rabbit,start,[normal,[]]},
                         {'EXIT',
                             {function_clause,
                                 [{autocluster,maybe_register,
                                      [error,aws,autocluster_aws],
                                      [{file,"src/autocluster.erl"},
                                       {line,111}]},
                                  {autocluster,init,0,
                                      [{file,"src/autocluster.erl"},
                                       {line,33}]},
                                  {rabbit_boot_steps,
                                      '-run_step/2-lc$^1/1-1-',1,
                                      [{file,"src/rabbit_boot_steps.erl"},
                                       {line,49}]},
                                  {rabbit_boot_steps,run_step,2,
                                      [{file,"src/rabbit_boot_steps.erl"},
                                       {line,49}]},
                                  {rabbit_boot_steps,
                                      '-run_boot_steps/1-lc$^0/1-0-',1,
                                      [{file,"src/rabbit_boot_steps.erl"},
                                       {line,26}]},
                                  {rabbit_boot_steps,run_boot_steps,1,
                                      [{file,"src/rabbit_boot_steps.erl"},
                                       {line,26}]},
                                  {rabbit,start,2,
                                      [{file,"src/rabbit.erl"},{line,585}]},
                                  {application_master,start_it_old,4,
                                      [{file,"application_master.erl"},
                                       {line,273}]}]}}}}
      in function  application_master:init/4 (application_master.erl, line 134)
    ancestors: [<0.150.0>]
    messages: [{'EXIT',<0.152.0>,normal}]
    links: [<0.150.0>,<0.31.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 2586
    stack_size: 27
    reductions: 98
  neighbours:
```

The 2nd to come up have nothing in the -sasl log  but does have this
rabbit@ip-10-0-1-212.eu-west-1.compute.internal.log

```
=INFO REPORT==== 21-Oct-2016::13:18:55 ===
Starting RabbitMQ 3.6.2 on Erlang 19.1.3
Copyright (C) 2007-2016 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 21-Oct-2016::13:18:55 ===
node           : rabbit@ip-10-0-1-212.eu-west-1.compute.internal
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
cookie hash    : jG/hKlMkavn9eqi0wsTy/A==
log            : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal.log
sasl log       : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal-sasl.log
database dir   : /var/lib/rabbitmq/mnesia/rabbit@ip-10-0-1-212.eu-west-1.compute.internal

=INFO REPORT==== 21-Oct-2016::13:18:58 ===
autocluster: Delaying startup for 249ms.

=INFO REPORT==== 21-Oct-2016::13:18:58 ===
autocluster: Starting aws registration.

=INFO REPORT==== 21-Oct-2016::13:18:59 ===
autocluster: Registering node with aws.

=ERROR REPORT==== 21-Oct-2016::13:18:59 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-212 is illegal **

=ERROR REPORT==== 21-Oct-2016::13:18:59 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-199 is illegal **

=ERROR REPORT==== 21-Oct-2016::13:18:59 ===
autocluster: Can not communicate with cluster nodes.
 
[and on and on]
```

And the 3rd to come up has logs very similar to the first

rabbit@ip-10-0-1-199.eu-west-1.compute.internal-sasl.log

```
=CRASH REPORT==== 21-Oct-2016::13:16:57 ===
  crasher:
    initial call: application_master:init/4
    pid: <0.151.0>
    registered_name: []
    exception exit: {bad_return,
                        {{rabbit,start,[normal,[]]},
                         {'EXIT',
                             {function_clause,
                                 [{lists,nth,
                                      [1,[]],
                                      [{file,"lists.erl"},{line,170}]},
                                  {autocluster_util,node_name_parse,3,
                                      [{file,"src/autocluster_util.erl"},
                                       {line,222}]},
                                  {autocluster_util,node_name,1,
                                      [{file,"src/autocluster_util.erl"},
                                       {line,173}]},
                                  {autocluster_aws,
                                      '-get_autoscaling_group_node_list/2-lc$^0/1-0-',
                                      1,
                                      [{file,"src/autocluster_aws.erl"},
                                       {line,117}]},
                                  {autocluster_aws,
                                      get_autoscaling_group_node_list,2,
                                      [{file,"src/autocluster_aws.erl"},
                                       {line,117}]},
                                  {autocluster,ensure_registered,2,
                                      [{file,"src/autocluster.erl"},
                                       {line,97}]},
                                  {autocluster,init,0,
                                      [{file,"src/autocluster.erl"},
                                       {line,33}]},
                                  {rabbit_boot_steps,
                                      '-run_step/2-lc$^1/1-1-',1,
                                      [{file,"src/rabbit_boot_steps.erl"},
                                       {line,49}]}]}}}}
      in function  application_master:init/4 (application_master.erl, line 134)
    ancestors: [<0.150.0>]
    messages: [{'EXIT',<0.152.0>,normal}]
    links: [<0.150.0>,<0.31.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 2586
    stack_size: 27
    reductions: 98
  neighbours:

```

rabbit@ip-10-0-1-199.eu-west-1.compute.internal.log

```
=INFO REPORT==== 21-Oct-2016::13:16:53 ===
Starting RabbitMQ 3.6.2 on Erlang 19.1.3
Copyright (C) 2007-2016 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 21-Oct-2016::13:16:53 ===
node           : rabbit@ip-10-0-1-199.eu-west-1.compute.internal
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
cookie hash    : jG/hKlMkavn9eqi0wsTy/A==
log            : /var/log/rabbitmq/rabbit@ip-10-0-1-199.eu-west-1.compute.internal.log
sasl log       : /var/log/rabbitmq/rabbit@ip-10-0-1-199.eu-west-1.compute.internal-sasl.log
database dir   : /var/lib/rabbitmq/mnesia/rabbit@ip-10-0-1-199.eu-west-1.compute.internal

=INFO REPORT==== 21-Oct-2016::13:16:55 ===
autocluster: Delaying startup for 994ms.

=INFO REPORT==== 21-Oct-2016::13:16:56 ===
autocluster: Starting aws registration.

=INFO REPORT==== 21-Oct-2016::13:16:57 ===
Error description:
   {could_not_start,rabbit,
       {function_clause,
           [{lists,nth,[1,[]],[{file,"lists.erl"},{line,170}]},
            {autocluster_util,node_name_parse,3,
                [{file,"src/autocluster_util.erl"},{line,222}]},
            {autocluster_util,node_name,1,
                [{file,"src/autocluster_util.erl"},{line,173}]},
            {autocluster_aws,'-get_autoscaling_group_node_list/2-lc$^0/1-0-',
                1,
                [{file,"src/autocluster_aws.erl"},{line,117}]},
            {autocluster_aws,get_autoscaling_group_node_list,2,
                [{file,"src/autocluster_aws.erl"},{line,117}]},
            {autocluster,ensure_registered,2,
                [{file,"src/autocluster.erl"},{line,97}]},
            {autocluster,init,0,[{file,"src/autocluster.erl"},{line,33}]},
            {rabbit_boot_steps,'-run_step/2-lc$^1/1-1-',1,
                [{file,"src/rabbit_boot_steps.erl"},{line,49}]}]}}
```

All help gratefully received.

**Comment by oli-logicnow** [comment]
Follow up info.

Rabbit did actually start on the 2nd instance to boot, but not the 1st or the 3rd.

Restarting with `sudo service rabbitmq-server start` worked on the 3rd (logs very similar to 2nd machine and no cluster).

Restarting with `sudo service rabbitmq-server start` failed on the 1st with the same logs as it had originally.

**Comment by binarin** [comment]
You should enable debug logs for autocluster. Then logs will contain information about http requests made to AWS API and about how autocluster interpeted this responses. Traces that you've provided show that autocluster_aws can't properly interpret response from AWS.

**Comment by oli-logicnow** [comment]
@binarin Excellent point, thank you.

I also noticed (2 minutes after posting here, naturally) that {aws_ec2_region, 'eu-west-1'} wasn't set.  Try this now.

**Comment by oli-logicnow** [comment]
Update to config

/etc/rabbitmq/rabbitmq.config
```
[
  {rabbit, [
    {log_levels, [{autocluster, debug}, {connection, debug}]}
  ]},
  {autocluster, [
    {backend, aws},
    {aws_autoscaling, true},
    {aws_ec2_region, 'eu-west-1'}
  ]}
].
```

Sadly  no additional errors logged.  All three instances did however start rabbit this time (one I killed of ip-10-0-1-183.eu-west-1.compute.internal.log, the instance that refused to start rabbit last time and let the AS do it's thing)

It looks like it's trying to use the wrong hostname to begin with

```
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-212 is illegal **
```



rabbit@ip-10-0-1-212.eu-west-1.compute.internal.log

```
=INFO REPORT==== 21-Oct-2016::14:47:51 ===
Starting RabbitMQ 3.6.2 on Erlang 19.1.3
Copyright (C) 2007-2016 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 21-Oct-2016::14:47:51 ===
node           : rabbit@ip-10-0-1-212.eu-west-1.compute.internal
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
cookie hash    : jG/hKlMkavn9eqi0wsTy/A==
log            : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal.log
sasl log       : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal-sasl.log
database dir   : /var/lib/rabbitmq/mnesia/rabbit@ip-10-0-1-212.eu-west-1.compute.internal

=INFO REPORT==== 21-Oct-2016::14:47:52 ===
autocluster: Delaying startup for 1476ms.

=INFO REPORT==== 21-Oct-2016::14:47:53 ===
autocluster: Starting aws registration.

=INFO REPORT==== 21-Oct-2016::14:47:54 ===
autocluster: Registering node with aws.

=ERROR REPORT==== 21-Oct-2016::14:47:54 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-212 is illegal **

=ERROR REPORT==== 21-Oct-2016::14:47:54 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-34 is illegal **

=ERROR REPORT==== 21-Oct-2016::14:47:54 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-199 is illegal **

=ERROR REPORT==== 21-Oct-2016::14:47:54 ===
autocluster: Can not communicate with cluster nodes.
```

**Comment by binarin** [comment]
You've probably tried to do it using instructions from wiki (`rabbit`, `log_levels`), but it appears that those instructions are outdated. You should try setting `autocluster_log_level` parameter of `autocluster` application.

**Comment by oli-logicnow** [comment]
@binarin You're racking up 🍺  rewards at a rate of knots here :D

**Comment by oli-logicnow** [comment]
Progress but no resolution.   It looks like the lookup find the hostnames in their short form and thus they don't resolve.  Time to read the aws lookup code I guess.

```
=INFO REPORT==== 21-Oct-2016::15:12:44 ===
node           : rabbit@ip-10-0-1-212.eu-west-1.compute.internal
home dir       : /var/lib/rabbitmq
config file(s) : /etc/rabbitmq/rabbitmq.config
cookie hash    : jG/hKlMkavn9eqi0wsTy/A==
log            : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal.log
sasl log       : /var/log/rabbitmq/rabbit@ip-10-0-1-212.eu-west-1.compute.internal-sasl.log
database dir   : /var/lib/rabbitmq/mnesia/rabbit@ip-10-0-1-212.eu-west-1.compute.internal

=INFO REPORT==== 21-Oct-2016::15:12:46 ===
autocluster: Using AWS backend

=INFO REPORT==== 21-Oct-2016::15:12:46 ===
autocluster: Delaying startup for 2687ms.

=INFO REPORT==== 21-Oct-2016::15:12:48 ===
autocluster: Starting aws registration.

=INFO REPORT==== 21-Oct-2016::15:12:48 ===
autocluster: Setting region: "eu-west-1"

=INFO REPORT==== 21-Oct-2016::15:12:49 ===
autocluster: Registering node with aws.

=INFO REPORT==== 21-Oct-2016::15:12:49 ===
autocluster: Registered node with aws.

=INFO REPORT==== 21-Oct-2016::15:12:49 ===
autocluster: Discovered ['rabbit@ip-10-0-1-212','rabbit@ip-10-0-1-199',
                         'rabbit@ip-10-0-1-34']

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-212 is illegal **

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-34 is illegal **

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-199 is illegal **
=INFO REPORT==== 21-Oct-2016::15:12:49 ===
autocluster: Discovered ['rabbit@ip-10-0-1-212','rabbit@ip-10-0-1-199',
                         'rabbit@ip-10-0-1-34']

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-212 is illegal **

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-34 is illegal **

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
** System running to use fully qualified hostnames **
** Hostname ip-10-0-1-199 is illegal **

=ERROR REPORT==== 21-Oct-2016::15:12:49 ===
autocluster: Can not communicate with cluster nodes.
```

**Comment by oli-logicnow** [comment]
```
autocluster: Discovered ['rabbit@ip-10-0-1-212','rabbit@ip-10-0-1-199',
                         'rabbit@ip-10-0-1-34']
```

comes from
[https://github.com/aweber/rabbitmq-autocluster/blob/a8271e8d71b38dd917957aee0f4bd35d055f43f6/src/autocluster.erl#L173](https://github.com/aweber/rabbitmq-autocluster/blob/a8271e8d71b38dd917957aee0f4bd35d055f43f6/src/autocluster.erl#L173)

I believe that  [https://github.com/aweber/rabbitmq-autocluster/blob/10bdd4c4f140f74ea9467095c7ed76a82c3d0414/src/autocluster_aws.erl#L196](https://github.com/aweber/rabbitmq-autocluster/blob/10bdd4c4f140f74ea9467095c7ed76a82c3d0414/src/autocluster_aws.erl#L196)  must be returning the wrong information.  I was distracted by `=INFO REPORT==== 24-Oct-2016::05:55:07 ===
autocluster: Registered node with aws.` being logged but I think this will always work with an AWS backend because of https://github.com/aweber/rabbitmq-autocluster/blob/10bdd4c4f140f74ea9467095c7ed76a82c3d0414/src/autocluster_aws.erl#L45


In short, still very stumped.  Has anybody else got Rabbit auto-clustering inside a private subnet of a VPC in AWS?

@binarin Please do send me an email address (and the cost of a local beer) to PayPal you the cost of a beer , without your help in correcting the debugging config I'd have not got this far.

**Comment by oli-logicnow** [comment]
Adding in specific hosts entries doesn't help here either.

```
10.0.1.185 ip-10-0-1-185
10.0.1.169 ip-10-0-1-169
10.0.1.179 ip-10-0-1-179
```

**Comment by binarin** [comment]
@oli-logicnow Please donate a few $ to EFF on my behalf =)

**Comment by oli-logicnow** [comment]
Done.
<img width="674" alt="screen shot 2016-10-24 at 12 30 19" src="https://cloud.githubusercontent.com/assets/17298623/19644079/b9ae9b3a-99e5-11e6-8d35-da25a78ede7d.png">

**Comment by gmr** [comment]
I use autoclustering in a private subnet using an autoscaling group, but I do not use longnames.

**Comment by oli-logicnow** [comment]
@gmr Great to hear that it's working, I'd assumed it was perfectly possible and my setup is the problem.

Yes - I'm using the AWS provides DNS solution.  I think the tickbox in VPC setup is called "DNS resolution" and is set to yes.  The "DNS hostnames" box is also set.

**Comment by gmr** [comment]
I'll look into the longname support, but I'm not likely to have time until later this week. Can I suggest you try without longname support turned on for now and see if that works for you?

```RABBITMQ_USE_LONGNAME=false```

**Comment by oli-logicnow** [comment]
No problem.  I think I only enabled it because the Rabbit docs thought it was important on EC2.

**Comment by oli-logicnow** [comment]
@gmr SUCCESS!

`[{nodes,[{disc,['rabbit@ip-10-0-1-169','rabbit@ip-10-0-1-179',
                'rabbit@ip-10-0-1-185']}]},
 {running_nodes,['rabbit@ip-10-0-1-169','rabbit@ip-10-0-1-185',
                 'rabbit@ip-10-0-1-179']},
 {cluster_name,<<"rabbit@ip-10-0-1-169.eu-west-1.compute.internal">>},`

My beer bounty stands - let me know where to send a few dollars/pounds/euros.

**Comment by jgorman-exp** [comment]
@gmr I also ran into the problem using long names.  rabbitmq-env.conf has `USE_LONGNAME=true` but a build of the autocluster plugin from master as of today kept logging something like the following:

````
autocluster: Fetching autoscaling = DNS: ["ip-10-38-85-128.us-west-2.compute.internal"]

autocluster: Discovered ['rabbit@ip-10-38-85-128']
````

and then logs an error saying the short name can't be used since rabbit is running with long names.

I updated my build of the plugin to hardcode this:  https://github.com/aweber/rabbitmq-autocluster/blob/a8271e8d71b38dd917957aee0f4bd35d055f43f6/src/autocluster_util.erl#L200

from
````
node_name_parse(autocluster_config:get(longname), Value)
````
to
````
node_name_parse(true, Value)
````

and everything works perfectly now.  So maybe something's off with `autocluster_config:get(longname)`...?

**Comment by hrzbrg** [comment]
Unfortunately I still have the problem, that the nodes to not cluster. 
I am running RabbitMQ 3.6.6 with Erlang 19.1 in a container very similar to [gavinmroy/alpine-rabbitmq-autocluster](https://github.com/gmr/alpine-rabbitmq-autocluster) just updated. They run in a private VPC on AWS. 

```=INFO REPORT==== 5-Feb-2017::12:51:14 ===
Starting RabbitMQ 3.6.6 on Erlang 19.1
Copyright (C) 2007-2016 Pivotal Software, Inc.
Licensed under the MPL.  See http://www.rabbitmq.com/

=INFO REPORT==== 5-Feb-2017::12:51:14 ===
node           : rabbit@ip-10-143-132-74
home dir       : /var/lib/rabbitmq
config file(s) : /usr/lib/rabbitmq/etc/rabbitmq/rabbitmq.config
cookie hash    : iqG7DCBA+lxNNLQq/Y6efg==
log            : tty
sasl log       : tty
database dir   : /var/lib/rabbitmq/mnesia

=INFO REPORT==== 5-Feb-2017::12:51:16 ===
autocluster: log level set to debug

=INFO REPORT==== 5-Feb-2017::12:51:16 ===
autocluster: Using AWS backend

=INFO REPORT==== 5-Feb-2017::12:51:16 ===
autocluster: Delaying startup for 2475ms.

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Starting aws registration.

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Setting region: "eu-west-1"

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Fetching autoscaling = Group: "sandbox-rabbitmq-asg"

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Fetching autoscaling = Instances: ["i-054f6ec2a20956ab9",
                                                "i-04d0b390a9356de44"]

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Fetching autoscaling = DNS: ["ip-10-143-132-74.eu-west-1.compute.internal",
                                          "ip-10-143-134-133.eu-west-1.compute.internal"]

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Registering node with aws.

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Registered node with aws.

=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Discovered ['rabbit@ip-10-143-132-74.eu-west-1.compute.internal',
                         'rabbit@ip-10-143-134-133.eu-west-1.compute.internal']

=ERROR REPORT==== 5-Feb-2017::12:51:18 ===
** System NOT running to use fully qualified hostnames **
** Hostname ip-10-143-132-74.eu-west-1.compute.internal is illegal **

=ERROR REPORT==== 5-Feb-2017::12:51:18 ===
** System NOT running to use fully qualified hostnames **
** Hostname ip-10-143-134-133.eu-west-1.compute.internal is illegal **

=ERROR REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Can not communicate with cluster nodes.
```

I already tried `RABBITMQ_USE_LONGNAME=false` and `RABBITMQ_USE_LONGNAME=true`. I used the stock version of rabbitmq-autocluster and build rabbitmq-autocluster with the changes suggested by @jgorman-exp. Everytime with no success. I don't know what I am doing wrong. The security group between the instances is definitely open. I always end up with `autocluster: Can not communicate with cluster nodes.`.

**Comment by michaelklishin** [comment]
Please read the log messages. It has nothing to do with security groups. Your nodes (or at least some of them) are not configured to use fully qualified names:

```
=INFO REPORT==== 5-Feb-2017::12:51:18 ===
autocluster: Discovered ['rabbit@ip-10-143-132-74.eu-west-1.compute.internal',
                         'rabbit@ip-10-143-134-133.eu-west-1.compute.internal']

=ERROR REPORT==== 5-Feb-2017::12:51:18 ===
** System NOT running to use fully qualified hostnames **
** Hostname ip-10-143-132-74.eu-west-1.compute.internal is illegal **

=ERROR REPORT==== 5-Feb-2017::12:51:18 ===
** System NOT running to use fully qualified hostnames **
** Hostname ip-10-143-134-133.eu-west-1.compute.internal is illegal **
```

See [Hostname Resolution Requirements](http://www.rabbitmq.com/clustering.html).

**Comment by hrzbrg** [comment]
I got a little over my head while fiddling around with it. The issue was, that the VPC has a custom search domain configured. I got it to work by telling docker to use `eu-west-1.compute.internal` as search domain via `--dns-search=eu-west-1.compute.internal` in the `docker run` command.
